server/src/main/scala/za/co/absa/atum/server/api/controller/CheckpointController.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.atum.server.api.controller
+
+import za.co.absa.atum.model.dto.CheckpointDTO
+import za.co.absa.atum.server.api.exception.ServiceError
+import za.co.absa.atum.server.api.service.CheckpointService
+import za.co.absa.atum.server.model.{ErrorResponse, GeneralErrorResponse, InternalServerErrorResponse}
+import zio._
+import zio.macros.accessible
+
+@accessible
+trait CheckpointController {
+  def createCheckpoint(checkpointDTO: CheckpointDTO): IO[ErrorResponse, CheckpointDTO]
+}
+
+class CheckpointControllerImpl(checkpointService: CheckpointService) extends CheckpointController {
```

Comment:
Is it a usual ZIO pattern to have both the trait and the class in the same file?

Reply:
@yruslan 
I haven't encountered any specific guidelines related to ZIO for this particular matter. Personally, I lean towards keeping interfaces or traits in separate files. However, I've noticed that this practice isn't consistently followed in the Atum project. Perhaps having these interfaces in the same files might better illustrate how to construct ZIO services. We can definitely reconsider this approach in the future, especially if we bring in other implementations. I'm curious to hear your thoughts on this topic. What's your perspective?

Reply:
This is actually funny. I saw this approach first and Hector, who did it in Atum Agent.
I am personally not that big of a fan either, but didn't consider it serious enough at that stage of development. Then I saw you @salamonpavel  using it and thought, it's one of the conventions out there. So I didn't bring it up anymore.

if we agree we prefer the style of having them in separate files, as it seems, let's do it 😄

Reply:
I'm on the same page. Having them separately is probably just a habit for me over some time. Definitely not a big deal in Scala, and if it better illustrates how ZIO works it is a good reason. For me class name having the same name as train/class inside is more for 2 reasons:
- It also going to match .class files in target
- IntelliJ idea does not place good icon on files for which class and file name do not match 😆  :
  
![Screenshot 2024-02-09 at 12 44 54](https://github.com/AbsaOSS/atum-service/assets/4082463/2c3459e5-5e66-4feb-8a4e-9e4ca3d914c1)

So overall no strong opinions, especially for Atum, completely up to you.